src/components/Gallery/Gallery.js

```diff
@@ -1,94 +1,88 @@
-import React from 'react';
+// @ts-check
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import Carousel, { Modal, ModalGateway } from 'react-images';
 import { withTranslationContext } from '../../context';
+import ImageModal from './ImageModal';
 
 /**
- * Gallery - displays up to 6 images in a simple responsive grid with a lightbox to view the images.
+ * Gallery - displays up to 4 images in a simple responsive grid with a lightbox to view the images.
  * @example ../../docs/Gallery.md
- * @extends PureComponent
+ * @type import('types').Gallery
  */
-class Gallery extends React.PureComponent {
-  static propTypes = {
-    images: PropTypes.arrayOf(
-      PropTypes.shape({
-        /** Url of the image */
-        image_url: PropTypes.string,
-        /** Url of thumbnail of image */
-        thumb_url: PropTypes.string,
-      }),
-    ),
-  };
-
-  state = {
-    modalIsOpen: false,
-    currentIndex: 0,
-  };
+const Gallery = ({ images, t }) => {
+  const [index, setIndex] = useState(0);
+  const [modalIsOpen, setModalOpen] = useState(false);
 
-  toggleModal = (index) => {
-    this.setState((state) => ({
-      modalIsOpen: !state.modalIsOpen,
-      currentIndex: index,
-    }));
+  /**
+   * @param {number} selectedIndex Index of image clicked
+   */
+  const toggleModal = (selectedIndex) => {
+    if (modalIsOpen) {
+      setIndex(0);
+      setModalOpen(false);
+    } else {
+      setIndex(selectedIndex);
+      setModalOpen(true);
+    }
   };
 
-  render() {
-    const { images, t } = this.props;
-    const formattedArray = images.map((image) => ({
-      src: image.image_url || image.thumb_url,
-    }));
+  // @ts-ignore
+  const formattedArray = images.map((image) => ({
+    src: image.image_url || image.thumb_url,
+  }));
 
-    const squareClass = images.length > 3 ? 'str-chat__gallery--square' : '';
+  const squareClass = images.length > 3 ? 'str-chat__gallery--square' : '';
```

Comment:
a small improvement but this can be added directly to the className without having a variable for it